core/src/main/java/org/apache/iceberg/io/AppenderBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.iceberg.MetricsConfig;
+import org.apache.iceberg.Schema;
+
+/**
+ * Interface which should be implemented by the data file format implementations. The {@link
+ * AppenderBuilder} will be parametrized based on the user provided configuration and finally the
+ * {@link AppenderBuilder#build(AppenderBuilder.WriteMode)} method is used to generate the appender
+ * for the specific writer use-cases. The following input should be handled by the appender in the
+ * specific modes:
+ *
+ * <ul>
+ *   <li>The appender's engine specific input type
+ *       <ul>
+ *         <li>{@link AppenderBuilder.WriteMode#APPENDER}
+ *         <li>{@link AppenderBuilder.WriteMode#DATA_WRITER}
+ *         <li>{@link AppenderBuilder.WriteMode#EQUALITY_DELETE_WRITER}
+ *       </ul>
+ *   <li>{@link org.apache.iceberg.deletes.PositionDelete} where the type of the row is the
+ *       appender's engine specific input type
+ *       <ul>
+ *         <li>{@link AppenderBuilder.WriteMode#POSITION_DELETE_WRITER}
+ *         <li>{@link AppenderBuilder.WriteMode#POSITION_DELETE_WITH_ROW_WRITER}
+ *       </ul>
+ * </ul>
+ *
+ * @param <B> type returned by builder API to allow chained calls
+ * @param <E> the engine specific schema of the input data
+ */
+public interface AppenderBuilder<B extends AppenderBuilder<B, E>, E> {
+  /** Set the file schema. */
+  B schema(Schema newSchema);
+
+  /**
+   * Set a writer configuration property.
+   *
+   * <p>Write configuration affects writer behavior. To add file metadata properties, use {@link
+   * #meta(String, String)}.
```

Comment:
```suggestion
   * <p>Write configuration affects writer behavior.
```

We don't need to reference the other method, just define what this one does.  No need to cross-reference.

Reply:
Simplified the javadoc basd on the suggestion.

For the record, copied from the `InternalData` javadoc. See: https://github.com/apache/iceberg/blob/dac5622338f024e9e5a565b8358d0b865edb4e2f/core/src/main/java/org/apache/iceberg/InternalData.java#L108